src/test/java/mate/academy/bookshop/controller/OrderControllerTest.java

```diff
@@ -0,0 +1,506 @@
+                        NON_EXISTING_ID)
+                        .with(SecurityMockMvcRequestPostProcessors.user(user)))
+                .andExpect(status().isNotFound())
+                .andReturn();
+        String actual = result.getResolvedException().getMessage();
+
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("Get specific order item with valid parameters")
+    @Sql(scripts = "classpath:database/orders/create-orders.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    public void getOrderItemByOrderId_ValidParams_ReturnsOrderItemResponseDto() throws Exception {
+        Long orderId = 1L;
+        Long itemId = 1L;
+        OrderItemResponseDto expected = retrieveOrdersFormSqlScript().stream()
+                .filter(o -> o.id().equals(orderId))
+                .flatMap(o -> o.orderItems().stream())
+                .filter(i -> i.id().equals(itemId))
+                .findFirst()
+                .orElse(null);
+
+        MvcResult result = mockMvc.perform(get("/orders/{orderId}/items/{itemId}",
+                        orderId,
+                        itemId)
+                        .with(SecurityMockMvcRequestPostProcessors.user(user)))
+                .andExpect(status().isOk())
+                .andReturn();
+        OrderItemResponseDto actual = objectMapper.readValue(result
+                        .getResponse().getContentAsString(),
+                OrderItemResponseDto.class);
+
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+    }
+
+    private void setAuthentication(User user) {
+        UserDetails userDetails = withUsername(user.getEmail())
+                .password(user.getPassword())
+                .authorities(user.getRoles().stream()
+                        .map(role -> new SimpleGrantedAuthority(role.getName().name()))
+                        .toArray(SimpleGrantedAuthority[]::new))
+                .build();
+
+        UsernamePasswordAuthenticationToken authToken = new UsernamePasswordAuthenticationToken(
+                userDetails, null, userDetails.getAuthorities());
+
+        SecurityContextHolder.getContext().setAuthentication(authToken);
+    }
+
+    private void compareOrderItems(Set<OrderItemResponseDto> expected,
+                                   Set<OrderItemResponseDto> actual) {
+        assertTrue(expected.stream()
+                .allMatch(expectedItem -> actual.stream()
+                        .anyMatch(actualItem ->
+                                EqualsBuilder.reflectionEquals(expectedItem,
+                                        actualItem,
+                                        "id"))));
+    }
+
+    private OrderResponseDto createOrderResponseDto() {
+
+        OrderItemResponseDto itemResponseDto1 = new OrderItemResponseDto(
+                1L,
+                3L,
+                BigDecimal.valueOf(12.99),
+                2
+        );
```

Comment:
I'd recommend using a builder instead, it is useful when you have a lot of fields that are of similar types, or you can also use lombok's `@Accessors(chain = true)`, or customize your setters so that they are not void but return `this`, giving you the opportunity to chain setters, which is very handy. 
But all that is up to you